src/ClientResolver.php

```diff
@@ -412,6 +427,12 @@ public static function _apply_endpoint_provider(callable $value, array &$args)
             if (isset($result['signatureVersion'])) {
                 $args['config']['signature_version'] = $result['signatureVersion'];
             }
+            if (isset($result['signingRegion'])) {
+                $args['config']['signing_region'] = $result['signingRegion'];
```

Comment:
I believe that case would be handled, but I agree that there should be an explicit test for overriding signing_region and signing_name.